scripts/run_commands.sh

```diff
@@ -0,0 +1,101 @@
+#
+# This scripts runs on the ec2 instance. It is uploaded to the EC2 instance by
+# test_commands.sh
+# The necessary env variables will be set by test_commands.sh using the
+# command line arguments provided to it.
+#
+
+declare -i FAILURES
+FAILURES=0
+
+RED='\033[1;31m'
+GREEN='\033[1;32m'
+LRED='\033[0;31m'
+LGREEN='\033[0;32m'
+NC='\033[0m' # No Color
+
+expect_success() {
+	if "${@}" &>> ~/ecs-cli-test-results/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${@}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${@}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+# used for commands that whose arguments might contain sensitive information
+expect_success_no_log() {
+	if "${@}" &>> ~/ecs-cli-test-results/test_output.txt; then
+		echo -e "${GREEN}SUCCEEDED${NC}: ${LGREEN}${1} ${2}${NC}"
+	else
+		echo "-------"
+		echo -e "${RED}FAILED${NC}: ${LRED}${1} ${2}${NC}"
+		echo "-------"
+		FAILURES+=1
+	fi
+}
+
+rm ~/ecs-cli-test-results/test_output.txt # clean up from past tests (in case this instance has been used before)
+# create the results directory and file; in case this instance is being used for the first time
+mkdir ~/ecs-cli-test-results/
+touch ~/ecs-cli-test-results/test_output.txt
+
+if ! [ -z "${gitname}" ]; then
+	# Not testing local changes, testing a branch instead
+	echo "Testing $branch"
+	# install git and go
+	echo "TYPE y|yes then enter to proceed."
```

Comment:
I can't seem to find where we are reading the answer of the prompt.

Reply:
The shell reads the user input itself. Specifically, the yum install command accepts the user input. The only reason I put in this print statement is so that the user knows that they have to type yes. Otherwise the install will not proceed (normally yum install will print a prompt to the terminal, but in the script I'm capturing the output and saving it to a file (so that extraneous output does swamp the user's terminal).

Reply:
You can use `yum -y install` to skip the prompt.  There's also the `yes` command that can be used for these kinds of things; you could also do `yes | yum install`.

Reply:
Thanks Sam!!